File: JavaScript/7js 3/tailSwap.js
/*
|--------------------------------------------------------------------------
| Tail Swap
|--------------------------------------------------------------------------
|
*/

function tailSwap(arr) {
   res = []
   for (let i = 0; i < arr.length; i++) {
      temp = arr[i].split(':')
      for (let j = 0; j < temp.length; j++) {
         res.push(temp[j]);
      }
   }
   return [res[0] + ':' + res[3], res[2] + ':' + res[1]]
}


function tailSwap(arr) {
   return (arr + "").replace(/(.+:)(.+),(.+:)(.+)/, "$1$4,$3$2").split(",")
}


function tailSwap(arr) {
   const [aa, ab] = arr[0].split(":")
   const [ba, bb] = arr[1].split(":")
   return [`${aa}:${bb}`, `${ba}:${ab}`]
}


function tailSwap(arr) {
   const [a, b] = arr.map(el => el.split(":"));
   return [`${a[0]}:${b[1]}`, `${b[0]}:${a[1]}`];
}


console.log(tailSwap(['abc:123', 'cde:456']), ['abc:456', 'cde:123']);
console.log(tailSwap(['a:12345', '777:xyz']), ['a:xyz', '777:12345']);

File: Python/7py 3/tail_swap.py
"""
|--------------------------------------------------------------------------
| Tail Swap
|--------------------------------------------------------------------------
|
"""

def tail_swap(strings):
   a1, a2 = strings[0].split(':')
   b1, b2 = strings[1].split(':')
   return [f'{a1}:{b2}',f'{b1}:{a2}']


def tail_swap(arr):
   fmt = '{}:{}'.format
   (head, tail), (head_2, tail_2) = (a.split(':') for a in arr)
   return [fmt(head, tail_2), fmt(head_2, tail)]


def tail_swap(strings):
   return '{0}:{3} {2}:{1}'.format(*[j for i in strings for j in i.split(':')]).split()


def tail_swap(lst):
   (a,b),(c,d) = (s.split(':') for s in lst)
   return list(map(':'.join, ((a,d),(c,b))))


print(tail_swap(['abc:123', 'cde:456']), ['abc:456', 'cde:123'])
print(tail_swap(['a:12345', '777:xyz']), ['a:xyz',   '777:12345'])